File: utils/gherkin.helpers.ts
export function parseGherkinTable<T extends Record<string, string>>(
  rows: T[],
): T {
  return rows[0];
}

export function tagScenario(tags: string[]): boolean {
  if (process.env.TAGS) {
    const filter = process.env.TAGS.split(',').map((t) => t.trim());
    return tags.some((tag) => filter.includes(tag.replace('@', '')));
  }
  return true;
}

File: utils/gherkin_helpers.py
def parse_gherkin_table(rows: list[dict[str, str]]) -> dict[str, str]:
    return rows[0]


def tag_scenario(tags: list[str]) -> bool:
    import os

    filter_tags = os.environ.get("TAGS")
    if filter_tags:
        allowed = [tag.strip() for tag in filter_tags.split(",")]
        return any(tag.replace("@", "") in allowed for tag in tags)
    return True
